src/test/java/net/datafaker/RandomFakerTest.java

```diff
@@ -20,47 +17,47 @@ public void before() {
         faker = new Faker(random);
     }
 
-    @Test
-    void testNumerifyRandomnessCanBeControlled() {
-        resetRandomSeed();
-        final String firstInvocation = faker.numerify("###");
-
-        resetRandomSeed();
-        final String secondInvocation = faker.numerify("###");
-        assertThat(firstInvocation).isEqualTo(secondInvocation);
-    }
-
-    @Test
-    void testLetterifyRandomnessCanBeControlled() {
-        resetRandomSeed();
-        final String firstInvocation = faker.letterify("???");
-
-        resetRandomSeed();
-        final String secondInvocation = faker.letterify("???");
-        assertThat(firstInvocation).isEqualTo(secondInvocation);
-    }
-
-    @Test
-    void testNameRandomnessCanBeControlled() {
-        resetRandomSeed();
-        final String firstInvocation = faker.name().name();
-
-        resetRandomSeed();
-        final String secondInvocation = faker.name().name();
-        assertThat(firstInvocation).isEqualTo(secondInvocation);
-    }
-
-    @Test
-    void testEmailRandomnessCanBeControlled() {
-        resetRandomSeed();
-        final String firstInvocation = faker.internet().emailAddress();
-
-        resetRandomSeed();
-        final String secondInvocation = faker.internet().emailAddress();
-        assertThat(firstInvocation).isEqualTo(secondInvocation);
-    }
-
-    private void resetRandomSeed() {
-        random.setSeed(CONSTANT_SEED_VALUE);
-    }
+//    @Test
```

Comment:
Commented out as I was questioning whether being able to change the seed by using the referenced object is a desirable thing you'd want to support. That seems like it could simply lead to unexpected behavior. The new approach allows for a new faker instance to be acquired with whatever seed is desired. 

An alternative would be to add a setSeed method inside RandomService to give someone the option to change the seed, while making the action more explicit in intent.